### PR TITLE
Added procedure to clear authentication and authorization cache

### PR DESCRIPTION
**Pull request by OliviaYtterbrink**
Added procedure to allow an administrator to force clearing of all cached authentication and authorization information.

changelog: Added procedure to clear authentication and authorization cache so admin can accelerate changes from cached providers like LDAP

### CODE REVIEW

File: enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/SecuritySettings.java [hunk 1]
```diff
@@ -100,6 +100,19 @@
...
+    @Description( "Determines if the result of authentication via the LDAP server should be cached or not. " +
+                  "Caching is used to limit the number of LDAP requests that have to be made over the network " +
+                  "for users that have already been authenticated successfully. A user can be authenticated against " +
+                  "an existing cache entry (instead of via an LDAP server) as long as it is alive " +
+                  "(see `dbms.security.realms.auth_cache_ttl`).\n" +
+                  "An important consequence of setting this to `true` than needs to be well understood, is that " +
+                  "Neo4j then needs to cache a hashed version of the credentials in order to perform credentials " +
+                  "matching. This hashing is done using a cryptographic hash function together with a random salt. " +
+                  "Preferably a conscious decision should be made if this method is considered acceptable by " +
+                  "the security standards of the organization in which this Neo4j instance is deployed." )
```
Comment (fickludd) [inline enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/SecuritySettings.java]: Nice description!

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresInteractionTestBase.java [hunk 2]
```diff
@@ -871,6 +870,23 @@ public void shouldNotListUsersForRoleIfNotAdmin() throws Exception
         testFailListRoleUsers( schemaSubject, ADMIN, PERMISSION_DENIED );
     }
 
+    //---------- clearing authentication cache -----------
+
+    @Test
+    public void shouldClearAuthenticationCacheIfAdmin() throws Exception
```
Comment (fickludd) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthProceduresInteractionTestBase.java]: `shouldHaveClearAuthenticationCacheProc`? It does not actually check the caching

File: enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java [hunk 3]
```diff
@@ -405,6 +405,13 @@ public void deleteRole( @Name( "roleName" ) String roleName )
         }
     }
 
+    @Description( "Clears authentication cache." )
+    @Procedure( name = "dbms.security.clearAuthenticationCache", mode = DBMS )
+    public void clearAuthenticationCache()
```
Comment (fickludd) [inline enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java]: I though we would be clearing to authorization cache too, although I forgot to put that into the card. Whaddayathink?

Comment (OliviaYtterbrink) [reply enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java]: Yes we should.

File: enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/integration/bolt/LdapAuthenticationIT.java [hunk 4]
```diff
@@ -595,6 +603,82 @@ public void shouldBeAbleToLoginWithLdapWhenSelectingRealmFromClient() throws Thr
         testAuthWithPublisherUser("tank", "ldap");
     }
 
+    @Test
+    public void shouldClearAuthenticationCache() throws Throwable
+    {
+        getLdapServer().setConfidentialityRequired( true );
+
+        try ( EmbeddedTestCertificates ignore = new EmbeddedTestCertificates() )
```
Comment (craigtaverner) [inline enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/integration/bolt/LdapAuthenticationIT.java:611]: Why test only with certificates? Would it not be easier to test without?

**Pull request merged**